### PR TITLE
[iOS] Fixed RadioButton text related properties not working

### LINKED ISSUE #19081: [iOS] RadioButton text related properties not working
### Description

Some of the properties related to the text style of `RadioButton` seem to have no effects on iOS.

Those I'm aware of are:

- `TextColor`
- `FontSize`
- `FontAttributes`

### Steps to Reproduce

1. Create a `RadioButton`
2. Try setting the properties listed above

For example
```xaml
<RadioButton
    x:Name="BuggedRadioButton"
    TextColor="Red"
    FontSize="50"
    TextTransform="Uppercase"
    FontAttributes="Bold"
    Content="Hello, world!" />
```

Actual result
Screenshot 2023-11-28 alle 16 35 20

Expected result
Screenshot 2023-11-28 alle 16 36 38

### Link to public reproduction project repository

https://github.com/Riccardo11/MauiIssues/tree/main/RadioButtonText

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

_No response_

### Did you find any workaround?

iOS support using a `View` object for its `Content` property, so you can use a `Label` that will be styled.

```csharp
#if IOS
BuggedRadioButton.Content = new Label
{
    Text = "Hello, world!",
    TextColor = Colors.Red,
    FontSize = 50,
    TextTransform = TextTransform.Uppercase,
    FontAttributes = FontAttributes.Bold
};
#endif
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

When the content was set as plain text, the label created based on that text never applied any text properties.

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/19081

|Demo|
|----|
|<video src="https://github.com/dotnet/maui/assets/42434498/36d93f56-bb29-4d6f-99ef-405d2887e88e"/>|

### CODE REVIEW

File: src/Controls/src/Core/RadioButton/RadioButton.iOS.cs [hunk 1]
```diff
@@ -8,10 +8,44 @@ public static void MapContent(RadioButtonHandler handler, RadioButton radioButto
 
 		public static void MapContent(IRadioButtonHandler handler, RadioButton radioButton)
 		{
+			// We want to modify the text in the content property only if
+			// the content is a string. In that case we want to apply
+			// the text values from the radio button
+			if (radioButton.Content is string
+				&& handler.VirtualView.PresentedContent is Element element
+				&& FindLabel(element) is Label label)
+			{
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/RadioButton/RadioButton.iOS.cs:17]: With the changes in the Mapper, are not similar changes required on Android as well?

Comment (kubaflo) [reply src/Controls/src/Core/RadioButton/RadioButton.iOS.cs:17]: Android radio buttons work without any changes needed. Therefore, I used the preprocessor directives `#if IOS` to modify only the iOS's mapper

File: src/Controls/tests/UITests/Tests/Issues/Issue19081.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+		[Test]
+		public void Issue19081Test()
+		{
+			this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.Android, TestDevice.Windows });
+
+			_ = App.WaitForElement("radioButton");
+
+			//Both radio buttons should look the same
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue19081.cs:23]: This test requires a reference snapshot. Waiting CI to generate it and include in a new commit.

Comment (jsuarezruiz) [reply src/Controls/tests/UITests/Tests/Issues/Issue19081.cs:23]: Added pending snapshot.

**Review by dustin-wojciechowski — DISMISSED** [review]

File: src/Controls/src/Core/RadioButton/RadioButton.iOS.cs [hunk 3]
```diff
@@ -8,10 +8,44 @@ public static void MapContent(RadioButtonHandler handler, RadioButton radioButto
 
 		public static void MapContent(IRadioButtonHandler handler, RadioButton radioButton)
 		{
+			// We want to modify the text in the content property only if
+			// the content is a string. In that case we want to apply
+			// the text values from the radio button
+			if (radioButton.Content is string
```
Comment (PureWeen) [inline src/Controls/src/Core/RadioButton/RadioButton.iOS.cs:14]: Can we resolve this by adding all these properties to templated bindings?

https://github.com/dotnet/maui/blob/b2870c7be752ccf8fa4085a45d28ef2d3200a8f1/src/Controls/src/Core/RadioButton/RadioButton.cs#L462C4-L464

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Comment by dustin-wojciechowski** [comment]
I pulled the latest from main and tested it against your UI Test and it looks like this problem may have already been fixed. See screenshot below: 
Simulator Screenshot - iPhone 15 - 2024-05-13 at 16 28 24

**Comment by kubaflo** [comment]
@dustin-wojciechowski I've also tested it and have the same observations. Therefore, I'm closing this PR

**Pull request closed without merging**